### PR TITLE
Take survey

### PR DESCRIPTION
**Pull request by lyndseysc**
functionality for user stories S2 S3 and S22

### CODE REVIEW

File: client/src/pages/TakeSurvey/SurveyDescription/SurveyDescription.jsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    <Box mt={4} alignItems='center' justifyContent='space-between'>
+      <Box mb={2}>
+        <Typography data-testid='survey-title' variant='h3'>
+          {survey.title}
+        </Typography>
+      </Box>
+      <Typography data-testid='survey-description' variant='h5'>
+        {survey.description}
+      </Typography>
+      {/* <Typography data-testid="survey-disclaimer" variant='h5'>{survey.disclaimer}</Typography> */}
```
Comment (marthaklambert) [inline client/src/pages/TakeSurvey/SurveyDescription/SurveyDescription.jsx]: remove comments pls

File: client/src/pages/TakeSurvey/SurveyDescription/SurveyDescription.test.jsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+describe('testing the survey description component', () => {
+  it(' test the description,title and disclaimer are rendered', () => {
+    const { getByTestId } = render(
+      <Provider store={store}>
+        <SurveyDescription />
+      </Provider>,
+    );
+    const SurveyDescriptionText = getByTestId('survey-description');
+    const SurveyTitleText = getByTestId('survey-title');
+    //  const SurveyDisclaimerText = getByTestId("survey-disclaimer)
```
Comment (marthaklambert) [inline client/src/pages/TakeSurvey/SurveyDescription/SurveyDescription.test.jsx]: remove comments pls

File: client/src/pages/TakeSurvey/SurveyDescription/SurveyDescription.test.jsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import { Provider } from 'react-redux';
+import { render } from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+import SurveyDescription from './SurveyDescription';
+import store from '../../../store/index';
+
+describe('testing the survey description component', () => {
```
Comment (marthaklambert) [inline client/src/pages/TakeSurvey/SurveyDescription/SurveyDescription.test.jsx]: not sure if describe makes sense with 1 test? I don't know though

File: client/src/pages/TakeSurvey/TakeSurvey.test.jsx [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+        <TakeSurvey />
+      </Provider>,
+    );
+
+    const PreviousButton = getByTestId('previous-button');
+
+    fireEvent.click(PreviousButton);
+    const SurveyDescriptionText = getByTestId('survey-description');
+    const SurveyTitleText = getByTestId('survey-title');
+    //  const SurveyDisclaimerText = getByTestId("survey-disclaimer)
```
Comment (marthaklambert) [inline client/src/pages/TakeSurvey/TakeSurvey.test.jsx]: comments

File: client/src/pages/TakeSurvey/index.jsx [hunk 5]
```diff
@@ -1,7 +1,41 @@
...
+import SurveyDescription from './SurveyDescription/SurveyDescription';
+import SurveySubmit from './SurveySubmit/SurveySubmit';
+import { useDispatch, useSelector } from 'react-redux';
+import { Box } from '@material-ui/core';
+import axios from 'axios';
 
 const TakeSurvey = () => {
-    return <></>;
+  const dispatch = useDispatch();
+  //const survey = useSelector((state) => state.takeSurveyReducer.survey);
```
Comment (marthaklambert) [inline client/src/pages/TakeSurvey/index.jsx]: comments

File: server/.env [hunk 6]
```diff
@@ -1,3 +1,6 @@
+
```
Comment (marthaklambert) [inline server/.env]: shouldn't this be in git ignore????

File: server/handlers/patchSurveys.js [hunk 7]
```diff
@@ -0,0 +1,30 @@
+const Joi = require('@hapi/joi');
+const addResponse = require('../queries/addResponse');
+const patchSurveys = async (req, res) => {
+  //TODO proper joi validation
```
Comment (marthaklambert) [inline server/handlers/patchSurveys.js]: when is this being done

File: server/queries/addResponse.js [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+    );
+
+    console.log('Response added to DB!');
+    const updateSuccessful = queryReturn.result.ok === 1;
+    if (!updateSuccessful) {
+      console.log('Query unsuccessful!');
+      return new Error('Query unsuccessful');
+    }
+  } catch (e) {
+    console.log('Oh no! An error!');
```
Comment (marthaklambert) [inline server/queries/addResponse.js]: can we remove this please, or do a better message

File: server/handlers/getSurveyAndQuestions.js [hunk 9]
```diff
@@ -0,0 +1,14 @@
+const readSurveyAndQuestions = require('../queries/readSurveyAndQuestions');
+
+const getSurveyAndQuestions = async (req, res) => {
+  const surveyId = req.params.id;
+  try {
+    let survey = await readSurveyAndQuestions(surveyId);
```
Comment (TKostrzewski) [inline server/handlers/getSurveyAndQuestions.js]: Could survey be a const?

File: server/handlers/patchSurveys.js (hunk 7 above)
Comment (lyndseysc) [reply server/handlers/patchSurveys.js]: @thomas-galligan

File: server/queries/addResponse.js (hunk 8 above)
Comment (lyndseysc) [reply server/queries/addResponse.js]: @thomas-galligan

Comment (TKostrzewski) [reply server/queries/addResponse.js]: Can we change 'e' to 'err' please?

Comment (thomas-galligan) [reply server/queries/addResponse.js]: both fixes are done

File: server/handlers/patchSurveys.js (hunk 7 above)
Comment (thomas-galligan) [reply server/handlers/patchSurveys.js]: 4 minutes ago

File: server/.env [hunk 10]
```diff
@@ -1,6 +1,3 @@
-
 JWT_SECRET=SECRET
 NODE_ENV=development
```
Comment (marthaklambert) [inline server/.env]: can we work out why this is in the repo please

Comment (thomas-galligan) [reply server/.env]: no idea since it's in all of the .gitignores

Comment (thomas-galligan) [reply server/.env]: removed it now

File: client/src/components/Main/Main.jsx [hunk 11]
```diff
@@ -22,7 +22,7 @@ const Main = () => {
           <Route exact path='/admin' component={Dashboard} />
           <Route exact path='/admin/surveys/create' component={CreateSurvey} />
           <Route path='/admin/surveys/:id' component={SurveyDetail} />
-          <Route path='/:id' component={TakeSurvey} />
+          <Route path='/takesurvey' component={TakeSurvey} />
```
[the remark this answers is not in the document]
Comment (lyndseysc) [reply client/src/components/Main/Main.jsx]: yes that's coming with personalised links

**Pull request merged**